src/tests/test_dal_middleware.py

```diff
@@ -0,0 +1,303 @@
+    assert isinstance(dal._reversed_middleware[0], FooMiddleware)
+
+
+def test_dal_middleware_custom_middleware():
+    """
+    Verify determining all middleware for the DAL to manage will
+    use our default middleware if none is specified, AND if any is
+    specified, our default middleware will be added as the last
+    middleware.
+
+    We also need to verify that these middlewares chain each other
+    in the order that we expect (reversed) and that they ultimately
+    use the default handler for dal methods.
+    """
+    # Verify default behavior first.
+    # Passing an empty DM here because we don't care about this
+    # for the purposes of the test, and it's not used.
+    dal = DataAccessLayer(data_manager=None)
+    assert len(dal._reversed_middleware) == 1
+    assert isinstance(dal._reversed_middleware[0], DalMethodResolverMiddleware)
+
+    # Verify custom default middleware
+    class ExampleMiddleware(MethodMiddleware):
+        pass
+
+    dal = DataAccessLayer(data_manager=None, default_middleware=(ExampleMiddleware,))
+    assert len(dal._reversed_middleware) == 1
+    assert isinstance(dal._reversed_middleware[0], ExampleMiddleware)
+
+    # Verify that even if you specify middleware, the default
+    # middleware gets appended to the end of the middleware stack.
+    # We test for this specifically because the default middleware
+    # needs to be run 'last'.
+
+    class FooMiddleware(MethodMiddleware):
+        pass
+
+    class BarMiddleware(MethodMiddleware):
+        pass
+
+    class BazMiddleware(MethodMiddleware):
+        pass
+
+    middlewares = [FooMiddleware, BarMiddleware, BazMiddleware]
+    dal = DataAccessLayer(data_manager=None, middleware=list(middlewares))
+
+    # The DAL handles making sure all of these are instantiated, but
+    # we have to manually do it for testing here.
+    middlewares.append(DalMethodResolverMiddleware)
+    expected_middlewares = []
+    for m in reversed(middlewares):
+        expected_middlewares.append(m())
+
+    # Verify if the objects in dal._reversed_middleware are instances of the
+    # expected middleware types in the right order
+    for i in range(0, len(expected_middlewares)):
+        assert isinstance(dal._reversed_middleware[0], type(expected_middlewares[0]))
+
+
+def test_dal_method_middleware():
+    """
+    Verify that when calling a method on a service off the DAL,
+    the method middlewares run, in the correct order and run
+    before the method itself is called.
+
+
+    We deliberately do not use the DalMethodResolverMiddleware here,
+    because we are testing middleware call chains, not functionality
+    of that class.
+    """
+    middleware_requests = []
+
+    def test_method():
+        middleware_requests.append(('default-handler-call', None))
+
+    class OuterMiddleware(MethodMiddleware):
+        def __call__(self, request, handler):
+            middleware_requests.append(('outer-middleware-ingress', request))
```

Comment:
You could take advantage of middleware and do something like `request.args += ('outer-middleware-ingress',)` and have your DAL method return the input args, then on the outbound have the middleware do `result += ('outer-middleware-egress',)` and verify all the middleware are represented there. That way you're validating middleware behavior and usage  more thoroughly.

So the actual dal method would be:

```
def example_method(*args):
     return args
```